From 12eeacb09af5c9596a5308c8bcad49745f05d210 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Thu, 19 Feb 2026 17:12:14 +0000
Subject: [PATCH 206/220] WIP: arm64: dts: rockchip: rk3588: Add IEP2 and IOMMU
 nodes

Add device nodes for the Image Enhancement Processor v2 (IEP2) at
0xfdbb0000 and its associated IOMMU at 0xfdbb0800 on RK3588. IEP2
is a hardware video deinterlacer in the VDPU power domain.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 arch/arm64/boot/dts/rockchip/rk3588-base.dtsi | 24 +++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
index 9d28682df84f..d8d1e36f29b0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
@@ -1402,6 +1402,30 @@ vepu121_3_mmu: iommu@fdbac800 {
 		#iommu-cells = <0>;
 	};
 
+	iep2: iep@fdbb0000 {
+		compatible = "rockchip,rk3588-iep2", "rockchip,rk3568-iep2";
+		reg = <0x0 0xfdbb0000 0x0 0x500>;
+		interrupts = <GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH 0>;
+		clocks = <&cru ACLK_IEP2P0>, <&cru HCLK_IEP2P0>,
+			 <&cru CLK_IEP2P0_CORE>;
+		clock-names = "aclk", "hclk", "sclk";
+		iommus = <&iep2_mmu>;
+		power-domains = <&power RK3588_PD_VDPU>;
+		resets = <&cru SRST_A_IEP2P0>, <&cru SRST_H_IEP2P0>,
+			 <&cru SRST_IEP2P0_CORE>;
+		reset-names = "axi", "ahb", "core";
+	};
+
+	iep2_mmu: iommu@fdbb0800 {
+		compatible = "rockchip,rk3588-iommu", "rockchip,rk3568-iommu";
+		reg = <0x0 0xfdbb0800 0x0 0x40>;
+		interrupts = <GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH 0>;
+		clocks = <&cru ACLK_IEP2P0>, <&cru HCLK_IEP2P0>;
+		clock-names = "aclk", "iface";
+		power-domains = <&power RK3588_PD_VDPU>;
+		#iommu-cells = <0>;
+	};
+
 	vdec0: video-codec@fdc38000 {
 		compatible = "rockchip,rk3588-vdec";
 		reg = <0x0 0xfdc38000 0x0 0x100>,
-- 
2.43.0

